lms/djangoapps/certificates/models.py

```diff
@@ -842,26 +842,43 @@ def course_key(self):
 class CertificateGenerationCourseSetting(TimeStampedModel):
     """Enable or disable certificate generation for a particular course.
 
-    This controls whether students are allowed to "self-generate"
-    certificates for a course.  It does NOT prevent us from
-    batch-generating certificates for a course using management
-    commands.
-
     In general, we should only enable self-generated certificates
     for a course once we successfully generate example certificates
     for the course.  This is enforced in the UI layer, but
     not in the data layer.
-
     """
     course_key = CourseKeyField(max_length=255, db_index=True)
-    enabled = models.BooleanField(default=False)
+    enabled = models.BooleanField(
+        default=False,
+        help_text=_(
+            u"DEPRECATED, please use self_generation_enabled instead."
+        )
+    )
+    # TODO: Learner-2549 remove deprecated enabled field
+
+    self_generation_enabled = models.BooleanField(
+        default=False,
```

Comment:
Since we're copying data from the old field to the new one, it might be useful to let this default to `null` to denote that it hasn't been explicitly set. That could be helpful if something goes wrong during the migration and we need to fix things.

Reply:
`None` is treated the same as `False` in Python. If something goes wrong with the migration, we aren't moving forward with the release. In fact, the migration will probably get reversed.

Reply:
Are you saying that if we default to `None`, it will save `FALSE` or `0` to the DB instead of `null`?

Reply:
MySQL will save either `NULL` or `FALSE`, depending on what you specify. Python will return either `None` or `False`. If you use `None`, you now have to add code to handle the `None` case.